### ISSUE TITLE
Add support for array of items of different types

### ISSUE THREAD

**Issue by nicolasrouanne** [jackwootton/json-schema#18]
When submitting a JSON with an array with different object types, the generator picks the first item and assumes the whole array consists of this kind of items.

Here is an example
```json
{
  "included": [
        {
            "id": "1",
            "type": "boat_model",
            "attributes": {
                "name": "Demacia",
                "boat_type": "sailboat",
                "loa": 16.63,
                "beam": 1.23,
                "draft": 1.74211606331017,
                "cabins": 5,
                "wc": 2
            }
        },
        {
            "id": "1",
            "type": "image",
            "attributes": {
                "url": "http://roberts.co/krystina",
                "main": true
            }
        }
  ]
}
```

There seems to be a solution in _JSON:Schema_ using `oneOf`; see [this issue](https://stackoverflow.com/a/15673308/5186918).

The generator should go through each element of arrays and check wether they are of the same type and add `oneOf` statements if not.

**Comment by jackwootton** [comment]
This feature has always been available. Just set the array options to "List Typing" rather than "Tuple Typing" (the default).

**Comment by nicolasrouanne** [comment]
@jackwootton OK, i figured it out. It seems I had to actually select "Tuple Typing" in the latest deployed version, and that "List Typing" is currently the default (display bug?). But it does enumerate the Array entities of different types.

Some notes though:
- I had to **manually remove _items of the same type in arrays_ in my sample JSON payload** to prevent ending up with a 10,000 lines JSON:Schema...
- I had the `Object > Required` option selected which is nice, but doesn't play well with `jsonapi` spec of empty collections for [linked resources](http://jsonapi.org/format/#document-resource-object-linkage)
  > Resource linkage **MUST** be represented as one of the following:
-`null` for empty to-one relationships.
-an empty array (`[]`) for empty to-many relationships.
-a single resource identifier object for non-empty to-one relationships.
-an array of resource identifier objects for non-empty to-many relationships.

I had to manually remove the `required` for _linked resources_ only.

Maybe I must have started saying I wanted to generate a _JSON:Schema_ for a JSON following `jsonapi` spec 🙃
